app/src/main/java/org/eu/droid_ng/wellbeing/lib/WellbeingFrameworkService.java

```diff
@@ -0,0 +1,141 @@
+                    onNullBinding(name);
+                    return;
+                }
+                wellbeingFrameworkService = Stub.asInterface(
+                        WellbeingFrameworkService.this.binder = service);
+                try {
+                    versionCode = wellbeingFrameworkService.versionCode();
+                } catch (Exception e) {
+                    Log.e("WellbeingFrameworkService", "Failed to get framework version", e);
+                    invalidateConnection();
+                    context.unbindService(this);
+                }
+                if (binder != null || initial) {
+                    notifyWellbeingService();
+                }
+            }
+
+            @Override
+            public void onServiceDisconnected(ComponentName name) {
+                invalidateConnection();
+                HANDLER.post(() -> tryConnect());
+            }
+
+            @Override
+            public void onBindingDied(ComponentName name) {
+                invalidateConnection();
+                context.unbindService(this);
+            }
+
+            @Override
+            public void onNullBinding(ComponentName name) {
+                invalidateConnection();
+                context.unbindService(this);
+                if (initial) {
+                    notifyWellbeingService();
+                }
+            }
+        };
+    }
+
+    private void invalidateConnection() {
+        wellbeingFrameworkService = DEFAULT;
+        versionCode = 0;
+        binder = null;
+    }
+
+    private void notifyWellbeingService() {
+        boolean initial = this.initial;
+        this.initial = false;
+        this.wellbeingService.onWellbeingFrameworkConnected(initial);
+    }
+
+    public void tryConnect() {
+        if (versionCode == -1) return;
+        if (binder == null || !(binder.isBinderAlive() && binder.pingBinder())) {
+            versionCode = -1;
+            try {
+                context.bindService(FRAMEWORK_SERVICE_INTENT, this.serviceConnection,
+                        Context.BIND_AUTO_CREATE | Context.BIND_INCLUDE_CAPABILITIES);
+            } catch (Exception e) {
+                Log.e("WellbeingFrameworkService", "Failed to bind framework service", e);
+                if (versionCode == -1) {
+                    versionCode = 0;
+                    if (initial) {
+                        notifyWellbeingService();
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public int versionCode() {
+        if (binder != null && !binder.isBinderAlive()) {
+            invalidateConnection();
+        }
+        // Allow partial emulation of the setAirplaneMode call if connection failed.
+        if (versionCode == 0 && system && !initial) {
+            return 1;
+        }
+        return versionCode;
+    }
+
+    @Override
+    public void setAirplaneMode(boolean value) throws RemoteException {
+        if (this.versionCode < 1) {
```

Comment:
this should either crash or do nothing

Reply:
I did this cause I though you would have liked having a fallback behavior.

Reply:
If it causes servere issues? No, thanks

Reply:
Well it's wifi/bluetooth not switching off, it's sever for the user.